gencode.c

```diff
@@ -8175,6 +8175,53 @@ gen_multicast(compiler_state_t *cstate, int proto)
 	/*NOTREACHED*/
 }
 
+struct block *
+gen_ifindex(compiler_state_t *cstate, int ifindex)
+{
+	register struct block *b0;
+
+	/*
+	 * Catch errors reported by us and routines below us, and return NULL
+	 * on an error.
+	 */
+	if (setjmp(cstate->top_ctx))
+		return (NULL);
+
+	/*
+	 * Only some data link types support ifindex qualifiers.
+	 */
+	switch (cstate->linktype) {
+	case DLT_LINUX_SLL2:
+		/* match packets on this interface */
+		b0 = gen_cmp(cstate, OR_LINKHDR, 4, BPF_W, ifindex);
+		break;
+        default:
+#if defined(linux) && defined(PF_PACKET) && defined(SO_ATTACH_FILTER)
```

Comment:
At this point, if we're building for Linux and neither `PF_PACKET` nor `SO_ATTACH_FILTER` are defined, libpcap won't build - we require a newer kernel - so it's sufficient to test whether `linux` is defined.

Reply:
Sure, I first posted this way before bfbe856 . Thanks for pointing this out.